pkg/migrations/sql/idempotent/maintenance.sql

```diff
@@ -0,0 +1,495 @@
+        PERFORM SCHEMA_CATALOG.delete_expired_series(metric_schema, metric_table, metric_series_table, ran_at, present_epoch, last_updated);
+        IF log_verbose THEN
+            RAISE LOG 'promscale maintenance: data retention: metric %: done deleting expired series as only action in %', metric_name, clock_timestamp()-lastT;
+            RAISE LOG 'promscale maintenance: data retention: metric %: finished in %', metric_name, clock_timestamp()-startT;
+        END IF;
+        RETURN;
+    END IF;
+
+    -- transaction 2
+        lastT := clock_timestamp();
+        PERFORM SCHEMA_CATALOG.set_app_name(format('promscale maintenance: data retention: metric %s: mark unused series', metric_name));
+        PERFORM SCHEMA_CATALOG.mark_unused_series(metric_schema, metric_table, metric_series_table, older_than, check_time);
+        IF log_verbose THEN
+            RAISE LOG 'promscale maintenance: data retention: metric %: done marking unused series in %', metric_name, clock_timestamp()-lastT;
+        END IF;
+    COMMIT;
+
+    -- transaction 3
+        lastT := clock_timestamp();
+        PERFORM SCHEMA_CATALOG.set_app_name( format('promscale maintenance: data retention: metric %s: drop chunks', metric_name));
+        PERFORM SCHEMA_CATALOG.drop_metric_chunk_data(metric_schema, metric_name, older_than);
+        IF log_verbose THEN
+            RAISE LOG 'promscale maintenance: data retention: metric %: done dropping chunks in %', metric_name, clock_timestamp()-lastT;
+        END IF;
+        SELECT current_epoch, last_update_time INTO present_epoch, last_updated FROM
+            SCHEMA_CATALOG.ids_epoch LIMIT 1;
+    COMMIT;
+
+
+    -- transaction 4
+        lastT := clock_timestamp();
+        PERFORM SCHEMA_CATALOG.set_app_name( format('promscale maintenance: data retention: metric %s: delete expired series', metric_name));
+        PERFORM SCHEMA_CATALOG.delete_expired_series(metric_schema, metric_table, metric_series_table, ran_at, present_epoch, last_updated);
+        IF log_verbose THEN
+            RAISE LOG 'promscale maintenance: data retention: metric %: done deleting expired series in %', metric_name, clock_timestamp()-lastT;
+            RAISE LOG 'promscale maintenance: data retention: metric %: finished in %', metric_name, clock_timestamp()-startT;
+        END IF;
+    RETURN;
+END
+$func$
+LANGUAGE PLPGSQL;
+GRANT EXECUTE ON PROCEDURE SCHEMA_CATALOG.drop_metric_chunks(text, text, timestamptz, timestamptz, boolean) TO prom_maintenance;
+
+CREATE OR REPLACE PROCEDURE SCHEMA_TRACING.drop_trace_chunks(_older_than timestamptz, log_verbose boolean = false)
+AS $func$
+DECLARE
+    _last timestamptz;
+    _start timestamptz;
+BEGIN
+    _start := clock_timestamp();
+
+    PERFORM SCHEMA_CATALOG.set_app_name('promscale maintenance: data retention: tracing');
+    IF log_verbose THEN
+        RAISE LOG 'promscale maintenance: data retention: tracing: starting';
+    END IF;
+
+    -- transaction 1
```

Comment:
I'd add an extra layer of safety such as if there is an error in any txn its caught and logged, but the rest of the actions continue. We've been caught by this before where an error in one part of the maintenance jobs makes the rest of the job not work.

Reply:
Good idea. Adding...